src/script_opt/ZAM/OPs/coercions.op

```diff
@@ -128,7 +128,9 @@ eval	if ( $1->Size() > 0 )
 unary-expr-op To-Any-Coerce
 op-type X
 set-type $1
-eval	AssignTarget($$, ZVal($1.ToVal(Z_TYPE), ZAM::any_base_type))
+eval	auto orig_lhs = $$; /* hold in case $$ = $1 */
+	$$ = ZVal($1.ToVal(Z_TYPE), ZAM::any_base_type);
+	ZVal::DeleteManagedType(orig_lhs);
```

Comment:
Would you be able to produce a test that reproduces this? I've tried a bunch, but only find unrelated issues.

The following crashes with `-O ZAM`, but not with normal script execution:
```
# @TEST-EXEC: zeek -b -O ZAM %INPUT >output
# @TEST-EXEC: btest-diff output

type X: record {
        a: string;
        b: any;
};

event zeek_init()
        {
        local x = X($a="123", $b=vector("abc"));
        print x;
        x$b = 1;
        print x;
        }
```


The following leaks memory with ZAM and normal execution:
```
# @TEST-EXEC: zeek -b -O ZAM %INPUT >output
# @TEST-EXEC: btest-diff output

type X: record {
        a: string;
};

event zeek_init()
        {
        local vec: vector of any;
        vec += X($a="abc-1");
        print vec;
        vec[0] = 1;
        print vec;
        }

```

Reply:
There's more. This one leaks memory:
```
# @TEST-EXEC: zeek -b -O ZAM %INPUT >output
# @TEST-EXEC: btest-diff output

module X;

export {
        option o: vector of string = vector();
}

event zeek_init()
        {
        local x: any = vector();
        x = (x as vector of string) + vector("aa");
        Config::set_value("X::o", "a" + "b");
        print X::o;
        print x;
        }


  =================================================================
  ==203873==ERROR: LeakSanitizer: detected memory leaks
  
  Direct leak of 24 byte(s) in 1 object(s) allocated from:
      #0 0x7f9987afe548 in operator new(unsigned long) ../../../../src/libsanitizer/asan/asan_new_delete.cpp:95
      #1 0x55b09c78c431 in vec_exec /home/awelzel/corelight-oss/zeek/src/script_opt/ZAM/ZBody.cc:624
      #2 0x55b09c7ac12b in zeek::detail::ZBody::Exec(zeek::detail::Frame*, zeek::detail::StmtFlowType&) /home/awelzel/corelight-oss/zeek/asan-build/src/ZAM-EvalDefs.h:274
      #3 0x55b09bce0699 in zeek::detail::ScriptFunc::Invoke(std::vector<zeek::IntrusivePtr<zeek::Val>, std::allocator<zeek::IntrusivePtr<zeek::Val> > >*, zeek::detail::Frame*) const /home/awelzel/corelight-oss/zeek/src/Func.cc:388
      #4 0x55b09baf2eb5 in zeek::EventHandler::Call(std::vector<zeek::IntrusivePtr<zeek::Val>, std::allocator<zeek::IntrusivePtr<zeek::Val> > >*, bool, double) /home/awelzel/corelight-oss/zeek/src/EventHandler.cc:96
      #5 0x55b09baef4da in zeek::Event::Dispatch(bool) /home/awelzel/corelight-oss/zeek/src/Event.cc:55
      #6 0x55b09baf0847 in zeek::EventMgr::Drain() /home/awelzel/corelight-oss/zeek/src/Event.cc:145
      #7 0x55b09b9b872a in zeek::detail::setup(int, char**, zeek::Options*) /home/awelzel/corelight-oss/zeek/src/zeek-setup.cc:1041
      #8 0x55b09c915c4d in main /home/awelzel/corelight-oss/zeek/src/main.cc:52
      #9 0x7f998122a1c9 in __libc_start_call_main ../sysdeps/nptl/libc_start_call_main.h:58
      #10 0x7f998122a28a in __libc_start_main_impl ../csu/libc-start.c:360
      #11 0x55b099fe5334 in _start (/home/awelzel/corelight-oss/zeek/asan-build/src/zeek+0xd65334) (BuildId: c33a6d858d4a47d8)
  
  SUMMARY: AddressSanitizer: 24 byte(s) leaked in 1 allocation(s).
```

And this one segfaults:
```
# @TEST-EXEC: zeek -b -O ZAM %INPUT >output
# @TEST-EXEC: btest-diff output

module X;

export {
        option o: vector of string = vector();
}

event zeek_init()
        {
        local x: any = vector();
        Config::set_value("X::o", vector("a") + (x as vector of string));
        print X::o;
        print x;
        }


  ==205583==ERROR: AddressSanitizer: SEGV on unknown address 0x000000000008 (pc 0x55b69ebec7fc bp 0x7fff2b31f830 sp 0x7fff2b31ecd0 T0)
  ==205583==The signal is caused by a READ memory access.
  ==205583==Hint: address points to the zero page.
      #0 0x55b69ebec7fc in std::_Optional_base_impl<zeek::ZVal, std::_Optional_base<zeek::ZVal, true, true> >::_M_is_engaged() const /usr/include/c++/13/optional:473
      #1 0x55b69ebec7fc in std::optional<zeek::ZVal>::operator bool() const /usr/include/c++/13/optional:987
      #2 0x55b69ebec7fc in vec_exec /home/awelzel/corelight-oss/zeek/src/script_opt/ZAM/ZBody.cc:628
      #3 0x55b69ec0c12b in zeek::detail::ZBody::Exec(zeek::detail::Frame*, zeek::detail::StmtFlowType&) /home/awelzel/corelight-oss/zeek/asan-build/src/ZAM-EvalDefs.h:274
      #4 0x55b69e140699 in zeek::detail::ScriptFunc::Invoke(std::vector<zeek::IntrusivePtr<zeek::Val>, std::allocator<zeek::IntrusivePtr<zeek::Val> > >*, zeek::detail::Frame*) const /home/awelzel/corelight-oss/zeek/src/Func.cc:388
      #5 0x55b69df52eb5 in zeek::EventHandler::Call(std::vector<zeek::IntrusivePtr<zeek::Val>, std::allocator<zeek::IntrusivePtr<zeek::Val> > >*, bool, double) /home/awelzel/corelight-oss/zeek/src/EventHandler.cc:96
      #6 0x55b69df4f4da in zeek::Event::Dispatch(bool) /home/awelzel/corelight-oss/zeek/src/Event.cc:55
      #7 0x55b69df50847 in zeek::EventMgr::Drain() /home/awelzel/corelight-oss/zeek/src/Event.cc:145
      #8 0x55b69de1872a in zeek::detail::setup(int, char**, zeek::Options*) /home/awelzel/corelight-oss/zeek/src/zeek-setup.cc:1041
      #9 0x55b69ed75c4d in main /home/awelzel/corelight-oss/zeek/src/main.cc:52
      #10 0x7fbc1442a1c9 in __libc_start_call_main ../sysdeps/nptl/libc_start_call_main.h:58
      #11 0x7fbc1442a28a in __libc_start_main_impl ../csu/libc-start.c:360
      #12 0x55b69c445334 in _start (/home/awelzel/corelight-oss/zeek/asan-build/src/zeek+0xd65334) (BuildId: c33a6d858d4a47d8)
```